pkg/apis/eksctl.io/v1alpha5/validation.go

```diff
@@ -430,6 +430,20 @@ func validateInstancesDistribution(ng *NodeGroup) error {
 		return fmt.Errorf("spotInstancePools should be between 1 and 20")
 	}
 
+	if distribution.SpotInstancePools != nil && distribution.SpotAllocationStrategy != nil && *distribution.SpotAllocationStrategy == SpotAllocationStrategyCapacityOptimized {
+		return fmt.Errorf("spotInstancePools cannot be specified when also specifying spotAllocationStrategy: %s", SpotAllocationStrategyCapacityOptimized)
+	}
+
+	if distribution.SpotAllocationStrategy != nil {
+		spotAllocationStrategies := make(map[string]bool)
```

Comment:
Check my earlier comment. I think for two strings the benefit of constructing a map for the look up is not worth it compared with the extra complexity of the code.

Reply:
👍